### DISCUSSION TITLE
Inovelli LZW36 sends 2 button press notifications

### DISCUSSION THREAD

**Discussion by gfischershaw** [zwave-js/zwave-js#2172] [General]
I was advised by Christian Moody of Inovelli that someone involved with Z-Wave JS had provided assistance to others having problems with multiple entities placed in the lifeline group for this switch, and that I might also get advice on this. I am using the standard HomeSeer Z-Wave Plug-in, and so honestly do not know why Inovelli pointed me in this direction, but maybe someone can help.
 
I am using an Inovelli LZW36 device under HomeSeer4 with Inovelli f/w v1.36 running on HomeSeer4. It automatically creates 2 associations when adding the device to the network. I deleted one of the associations, as it was causing problems, leaving only Group 1, with endpoint 1 pointing at HomeSeer.
 
When I press/release one of the two main buttons (not the small side toggles), it appears that two packets are sent rather than one, and indeed, the event handler listening to a button press is triggered twice. What I see in the log also confirms this:
 
ApplicationCommandHandler from Network: F255241A Node: 154 HANDLING: COMMAND_CLASS_MULTI_CHANNEL_V3 Frame(7)=13
Payload1: 5B-03-3F-80-01-B1-00-42
Frame: 60-0D-00-01-5B-03-3F-80-01-B1-00-42
Payload1 zd is not nothing
Set_New_Level_Real: Initial device is Main Floor Library Central Scene
Set_New_Level_Real called for Main Floor Library Central Scene, Type=COMMAND_CLASS_CENTRAL_SCENE_V2, EndPoint=0
Set_New_Level, Parent: Main Floor Library Scene.Root, Child:Main Floor Library Central Scene
Setting device Central Scene to value 1000 Network: F255241A Node: 154 Device Ref: 1890
Event Trigger "Main Floor Library.Cabinets - Toggle"
Payload1: 5B-03-3F-80-01-B1-00-42
Payload1 zd is not nothing
Set_New_Level_Real: Initial device is Main Floor Library Central Scene
Set_New_Level_Real called for Main Floor Library Central Scene, Type=COMMAND_CLASS_CENTRAL_SCENE_V2, EndPoint=0
Device: Main Floor Library Cabinets to Off (0) by/from: CAPI Control Handler
Set_New_Level, Parent: Main Floor Library Scene.Root, Child:Main Floor Library Central Scene
Setting device Central Scene to value 1000 Network: F255241A Node: 154 Device Ref: 1890
Event Trigger "Main Floor Library.Cabinets - Toggle"
 
So it looks like a V3 packet is translated into a V2 button press packet by the Plug-in and triggers the event handler. Then a V2 packet is received and triggers the event handler again,

**Comment by marcus-j-davies** [comment]
> I am using the standard HomeSeer Z-Wave Plug-in, and so honestly do not know why Inovelli pointed me in this direction,

I would have to agree here.
Whilst the devs involved with z-wave js have contacts within Inovelli, and many conversations have happened with them, they have been to discuss  work arounds developed in z-wave js for devices that may not follow specifications, or problematic devices. The work arounds are specific to z-wave js, and nothing else.

Sure, if someone does have some info, then happy days - there are some very kind folk here.

**Comment by AlCalzone** [comment]
I'm not sure how to read the HomeSeer logs and if these are really two packets, but the sequence number `3F` on both are the same. So either you're misinterpreting the log or HomeSeer is not filtering out duplicate frames. IMO this looks more like a COMMAND_CLASS_CENTRAL_SCENE_V2 that is encapsulated in a COMMAND_CLASS_MULTI_CHANNEL_V3, which would be expected with multi channel associations.

> It automatically creates 2 associations when adding the device to the network. I deleted one of the associations, as it was causing problems, leaving only Group 1, with endpoint 1 pointing at HomeSeer.

AFAIK, the standards-compliant way would be setting up a lifeline to the controller, but to endpoint 0, not 1. We've had some problems with devices where zwave-js set up the endpoint 0 association, but there was one left to endpoint 1 which OZW had set up and resulted in non-encapsulated and/or duplicate reports.

> **Multi Channel Association Command Class, version 3**
> [...]
> This version introduces support for the creation of an End Point Association to the Root Device of
another device, e.g. a gateway, by allowing the destination End Point 0 as a valid value.
This allows a gateway to create a single Lifeline association from a Multi Channel device and
subsequently receive status messages or sensor reports from the Root Device as well as the End Points
of that device. Refer to section 4.32.2.

**Reply by gfischershaw** [reply]
Very useful comments. When I change the association to endpoint 0, there no longer appears to be a duplicate frame. Only the V2 frame appears in the log, and the event handler is only called once.
